src/ui/map/TileSet.svelte

```diff
@@ -39,15 +39,9 @@
   }
 
   // watches for isMapLoaded then runs the addSource method
-  function isMapLoaded() {
-    if (map.isStyleLoaded(id)) {
-      addSource();
-    } else {
-      setTimeout(() => {
-        isMapLoaded();
-      }, 100);
-    }
-  }
+  map.on("style.load", () => {
```

Comment:
Let's stick a comment in there as per Pete's suggestion and ship it.

Reply:
Wicked, I'll sort that now.